src/foam/box/DedicatedWorkerRegistry.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'foam.box',
+  name: 'DedicatedWorkerRegistry',
+  extends: 'foam.box.ProxyBox',
+  implements: [ 'foam.box.BoxRegistryBox' ],
+
+  requires: [
+    'foam.core.StubFactorySingleton',
+    'foam.box.BoxRegistry',
+    'foam.box.node.ForkBox'
+  ],
+
+  imports: [ 'registry' ], // __context__.registry is the context it is declared in
+
+  exports: [ 'as registry' ], // __subContext__.registry === this
+
```

Comment:
I am not sure if this is the appropriate action. Previous, it was discussed that we would do something along the lines of
```
var ctx = foam.box.Context();
ctx.registry = foam.box.DedicatedWorkerRegistry.create({
  delegate: ctx.registry
}, ctx);
```
But this causes a cyclic reference to itself. If we create them separately
```
var ctx = foam.box.Context();
var registry = foam.box.DedicatedWorkerRegistry.create({
  delegate: ctx.registry
}, ctx);
```
then use `registry.register`, registrations will be ok, but issues will come up later on. An example:
```
var stub = foam.core.StubFactorySingleton.create().get(foam.dao.DAO)
   .create({ delegate: register }, ctx);
stub.put(...)
```
As this will register will a different registry, and then the stub call will not be able to find the box in the registry, (in the current implementation)